### PR TITLE
Add config file lookup on default location if not passed by user

### PR DESCRIPTION
**Pull request by rajkumarpatel2602**
* Add lookup for config.yaml on default location
* Add info on help message for user's config CLI option

Signed-off-by: Rajkumar Patel <rajkumar.patel@seagate.com>

### CODE REVIEW

File: s3/replication/manager/src/s3replicationmanager/__main__.py [hunk 1]
```diff
@@ -29,7 +29,9 @@ def setup_args(parser):
         '--configfile',
         type=str,
         metavar='path',
-        help='Path to configuration file(format: yaml)')
+        help='Path to configuration file(format: yaml).\n'
```
Comment (kaustubh-d) [inline s3/replication/manager/src/s3replicationmanager/__main__.py:32]: how about this?
```
help="""Path to configuration file(format: yaml).
e.g: ~/.cortxs3/credentials.yaml """
```

File: s3/replication/manager/src/s3replicationmanager/config.py [hunk 2]
```diff
@@ -29,10 +29,26 @@ class Config:
...
+                    "Config read from default location : {}".format(
+                        self.configfile))
+            else:
+                self.configfile = os.path.join(os.path.dirname(__file__),
+                                               '..', 'config', 'config.yaml')
+                print(
+                    "Config read from source : {}".format(
+                        self.configfile))
         else:
             self.configfile = configfile
```
Comment (kaustubh-d) [inline s3/replication/manager/src/s3replicationmanager/config.py:48]: `self.configfile = os.path.abspath(configfile)` so relative paths with `~` can also work

File: s3/replication/replicator/src/s3replicator/__main__.py [hunk 3]
```diff
@@ -29,7 +29,9 @@ def setup_args(parser):
         '--configfile',
         type=str,
         metavar='path',
-        help='Path to configuration file(format: yaml)')
+        help='Path to configuration file(format: yaml).\n'
```
Comment (kaustubh-d) [inline s3/replication/replicator/src/s3replicator/__main__.py:32]: Same as above

File: s3/replication/manager/src/s3replicationmanager/config.py (hunk 2 above)
Comment (kaustubh-d) [reply s3/replication/manager/src/s3replicationmanager/config.py:48]: `os.path.expanduser(configfile) `

File: s3/replication/manager/src/s3replicationmanager/config.py [hunk 4]
```diff
@@ -29,10 +29,26 @@ class Config:
...
         if configfile is None:
-            self.configfile = os.path.join(os.path.dirname(__file__),
-                                           '..', 'config', 'config.yaml')
+            # Default config file directory : ~/.cortxs3/manager
+            config_home = os.path.join(
+                os.path.expanduser('~'), '.cortxs3', 'manager')
+            configfile = os.path.join(config_home, 'config.yaml')
+            if os.path.isfile(configfile):
+                self.configfile = configfile
+                print(
```
Comment (kaustubh-d) [inline s3/replication/manager/src/s3replicationmanager/config.py:38]: you can just use one print at end after path is resolved.

File: s3/replication/manager/src/s3replicationmanager/__main__.py (hunk 1 above)
Comment (rajkumarpatel2602) [reply s3/replication/manager/src/s3replicationmanager/__main__.py:32]: Docstring containing both lines is not represented properly in output. so I am using, single quotes for each line.

Output would be as below :
image

File: s3/replication/manager/src/s3replicationmanager/config.py (hunk 2 above)
Comment (rajkumarpatel2602) [reply s3/replication/manager/src/s3replicationmanager/config.py:48]: done

File: s3/replication/replicator/src/s3replicator/__main__.py (hunk 3 above)
Comment (rajkumarpatel2602) [reply s3/replication/replicator/src/s3replicator/__main__.py:32]: done

File: s3/replication/manager/src/s3replicationmanager/config.py (hunk 4 above)
Comment (rajkumarpatel2602) [reply s3/replication/manager/src/s3replicationmanager/config.py:38]: done

**Review by kaustubh-d — APPROVED** [review]

**Pull request merged**